CONTRIBUTING.md

```diff
@@ -48,15 +48,25 @@ The current maintainers are listed in the [CODEOWNERS](.github/CODEOWNERS) file.
 
 If you are interested in being pinged when testing is needed for a certain platform, please add yourself to the [Testers and Contributors](https://github.com/rust-windowing/winit/wiki/Testers-and-Contributors) table!
 
-## Making a new release
-
-If you believe a new release is warranted, you can make a pull-request with:
-- An updated version number (remember to change the version everywhere it is used).
-- A new section in the changelog (below the `# Unreleased` section).
-
-This gives contributors an opportunity to squeeze in an extra PR or two that they feel is valuable
-enough to warrant blocking the release a little.
-
-Once the PR is merged, a maintainer will create a new tag matching the version name (e.g. `v0.26.1`),
-and a CI job will automatically release the new version. Remember that the release date in the
-changelog must be kept in check with the actual release date.
+## Release process
+
+Given that winit is a widely used library we should be able to make a patch
+releases at any time we want without blocking the development of new features.
+
+To achieve these goals, a new branch is created for every new release. Releases
+and later patch releases are committed and tagged in this branch.
+
+The exact steps for an exemplary `0.2.0` release might look like this:
+  1. Initially the version on the latest master is `0.1.0`
+  2. A new `v0.2.x` branch is created for the release
+  3. In the branch, the version is bumped to `v0.2.0`
+  4. The new commit in the branch is tagged `v0.2.0`
+  5. The version is pushed to crates.io
+  6. A GitHub release is created for the `v0.2.0` tag
+  7. On master, the version is bumped to `0.2.0` and the CHANGELOG is updated
```

Comment:
I'd like the commit from which the release was made (the commit hash is present in crates metadata) to be reachable from `master`

Reply:
Reachable in what way? It'll reside on its own branch created from master branch. You can't have it on master.

Reply:
Reachable as-in when I've checked out `master`, I can see the tag in `git log`

Reply:
Not having the release on `master` is kinda essential to this entire release process.

The point is not having to go for any roundtrips holding up master while making releases.

Reply:
@madsmtm you can `git checkout tag`. But yeah, the entire point is not having releases on master.

Reply:
What I wanted to say was, it would be nice if we could have the first `0.x.0` release be visible from `master`; of course the rest wouldn't be

Reply:
That means you won't be able to make any commits to master from RC start to RC end of that `0.x.0` release though. If you have longer release/testing processes, that can significantly hold back features from getting merged.

Imo no PR should ever be held back because you're in the process of making a release, to me that indicates the release process is broken.

Reply:
We won't have RCs for example and will have a branch for all ongoing releases. So we could tag it, but it's visible anyway, since you branched out from master.